src/App.tsx

```diff
@@ -1,21 +1,71 @@
-import './App.css'
 import CarForm from './components/carForm'
 import { useState } from 'react'
 import { Car } from './types'
+import VCV from '../lib/vcv'
 import CarList from './components/carList'
+import CarEditDialog from './components/carEditDialog'
+import CarHistory from './components/carsHistory'
+import { Box, Button, Container, Divider, ThemeProvider} from '@mui/material'
+
+import themes from './themes'
 
 function App() {
-  const [cars, setCars] = useState<Car[]>([]) // Car array state
+	const { state : cars, 
+					setValue : setCars, 
+					revertTo : revertCommit, 
+					history : commitHistory,
+					index: commitIndex } = VCV<Car[]>([]) // Car array state
+	const [themeIndex, setThemeIndex] = useState(0)
+	const [editDialogOpen, setEditDialogOpen] = useState(false)
+	const [editIndex, setEditIndex] = useState(-1)
+
```

Comment:
I think it will be more readable to initialize this state with `null` or `undefiend` instead of not existing index.
You can set the type using `useState<number | undefined` or `useState<number | null`